File: Backend/SecureBooking.Domain/Entities/Booking.cs
using System;
using SecureBooking.Shared.Enums;

namespace SecureBooking.Domain.Entities;

public class Booking : Entity
{
    public Guid UserId { get; set; }
    public User? User { get; set; }

    public Guid RoomId { get; set; }
    public Room? Room { get; set; }

    public DateTime CheckIn { get; set; }

    public DateTime CheckOut { get; set; }

    public BookingStatus Status { get; set; } = BookingStatus.Pending;

    public string? Notes { get; set; }
}

File: Frontend/secure-booking-web/src/types/Booking.ts
export enum BookingStatus {
  Pending = "Pending",
  Confirmed = "Confirmed",
  CheckedIn = "CheckedIn",
  CheckedOut = "CheckedOut",
  Cancelled = "Cancelled",
}

export interface Booking {
  id: string;
  userId: string;
  userFullName: string;
  userEmail: string;
  roomId: string;
  roomName: string;
  hotelName: string;
  checkIn: string; // ISO date string
  checkOut: string; // ISO date string
  status: BookingStatus;
  notes?: string;
  totalPrice: number;
  createdAt: string;
}

export interface CreateBookingRequest {
  roomId: string;
  checkIn: string;
  checkOut: string;
  notes?: string;
}
